src/adapter/http/src/smart_protocol/axum_server_push_protocol.rs

```diff
@@ -132,7 +132,10 @@ impl AxumServerPushProtocolInstance {
                                     PushPhase::InitialRequest,
                                 ))
                             })?;
-                            return Ok(());
+                            let mut notify_interval = tokio::time::interval(
+                                std::time::Duration::from_secs(MIN_UPLOAD_PROGRESS_PING_DELAY_SEC),
+                            );
+                            notify_interval.tick().await;
```

Comment:
If I understand the idea correctly, it is the desire to pass control to another asynchronous task?

> The first tick completes immediately 
(c) https://docs.rs/tokio/latest/tokio/time/struct.Interval.html

If that's the case, how about `tokio::task::yield_now`?

Reply:
Not really. I just give a time for a client to handle sent error message before closing a socket